### PR TITLE
Manual VMS update merge:

### PR DESCRIPTION
**Pull request by marcelino-a**
1. Refactoring of computation of vms stabilization parameter (tau)
2. Updated VMS edge bubble definition
3. Heuristic correction to tau for overlapping sectors (edge bubble support)
4.  Used stabilization parameter (tau) for jump norm of error when using vms

### CODE REVIEW

File: physics/Compadre_ReactionDiffusion_Operator.cpp [hunk 1]
```diff
@@ -575,20 +596,20 @@ void ReactionDiffusionPhysics::initialize() {
...
-
-        //Forming matrix Amat (for scalar problem, it is a scalar)
+        // COMPUTE VMS tau^(a)_s (stabilization tensor tau associated to side s and element a)
+        // Allocate matrix Amat (for scalar problem, it is a scalar)
         Teuchos :: SerialDenseMatrix <local_index_type, scalar_type> Amat(_ndim_requested, _ndim_requested);
         Teuchos :: SerialDenseSolver <local_index_type, scalar_type> Amat_solver;
         
-        //could parallelize outer two loops
+	// Allocate array of tau^(a)_s for all s, a
+        auto num_edges = _ndim_requested + 1; //MA: true for triangles and tetrahedra only
```
Comment (kuberry) [inline physics/Compadre_ReactionDiffusion_Operator.cpp:609]: Drop this line and use num_sides in following line.

File: examples/Compadre_ReactionDiffusion.cpp [hunk 2]
```diff
@@ -23,6 +23,13 @@
 #include <Compadre_ReactionDiffusion_BoundaryConditions.hpp>
 #include <Compadre_ReactionDiffusion_Operator.hpp>
 
+// ####################################
```
Comment (kuberry) [inline examples/Compadre_ReactionDiffusion.cpp:26]: Drop

Comment (kuberry) [reply examples/Compadre_ReactionDiffusion.cpp:26]: Add an assertion that it doesn't work for constant basis functions due to calculation of cubature_degree.

File: physics/Compadre_ReactionDiffusion_Operator.cpp [hunk 3]
```diff
@@ -465,19 +479,44 @@ void ReactionDiffusionPhysics::initialize() {
...
+        int bub_pow = input_poly_order/2 + 1;
+        double bub_coeff = 1.5;
+
+        // VMS Edge bubble VALUES on SIDES
+        std::vector<Intrepid::FieldContainer<double> > basis_T6_edge_values_pow(num_element_sides, Intrepid::FieldContainer<double>(num_basis_functions, num_side_cub_points) ); //allocate
+        int size_basis_T6_edge_values_1 = num_basis_functions * num_side_cub_points;
+        for (int i=0; i<num_element_sides; ++i) {
+    	    for (int k=0; k<size_basis_T6_edge_values_1; ++k) {
+                double b1 = basis_T6_edge_values[i][k]; //values at SIDE cubature points
+    	        basis_T6_edge_values_pow[i][k] = bub_coeff * std::pow(b1, bub_pow) + (1.0 - bub_coeff) * b1;
```
Comment (kuberry) [inline physics/Compadre_ReactionDiffusion_Operator.cpp:507]: Add comment of `heuristic` somewhere

**Comment by kuberry** [comment]
Hi @marcelino-a. I made some changes to remove some commented out code and add some of the comments that we talked about. Would you mind pulling these changes and testing again to make sure it didn't break anything?

**Comment by marcelino-a** [comment]
Thanks, Paul! I will pull and check the changes

**Comment by marcelino-a** [comment]
Hi @kuberry, I pulled your changes. I also added some other changes of my own, and checked that everything is working:
1.I removed the variable "vmsBCfactor" from physics/Compadre_ReactionDiffusion_Sources.cpp and examples/Compadre_ReactionDiffusion.cpp because it was a feature used for testing that is no longer needed.
2. In physics/Compadre_ReactionDiffusion_Operator.cpp, in the block that determines cubature_degree for VMS, I made the logic a bit more explicit, as we had discussed.
3. I verified that VMSDG still gives optimal convergence (used variational_convergence.py, --solution=sine, checked for order=1,2,3,4 and 4 meshes)

**Review by kuberry — APPROVED** [review]

**Comment by kuberry** [comment]
Looks great! Thanks @marcelino-a

**Pull request merged**